diff --git a/greengrass-opcua-adapter-nodejs/node_modules/node-opcua/lib/client/opcua_client.js b/greengrass-opcua-adapter-nodejs/node_modules/node-opcua/lib/client/opcua_client.js
index 875d850..437aa69 100644
--- a/greengrass-opcua-adapter-nodejs/node_modules/node-opcua/lib/client/opcua_client.js
+++ b/greengrass-opcua-adapter-nodejs/node_modules/node-opcua/lib/client/opcua_client.js
@@ -48,6 +48,7 @@ var utils = require("lib/misc/utils");
 var debugLog = utils.make_debugLog(__filename);
 var doDebug = utils.checkDebugFlag(__filename);
 
+var ConfigAgent = require('../../../../config_agent');
 
 function validateServerNonce(serverNonce) {
     return (serverNonce && serverNonce.length < 32) ? false : true;
@@ -239,6 +240,13 @@ OPCUAClient.prototype.__createSession_step2 = function (session,callback) {
                 self._server_endpoints = response.serverEndpoints;
                 session.serverEndpoints = response.serverEndpoints;
 
+                var result = ConfigAgent.compareWithTrustCert(session.serverCertificate.toString("base64"));
+                console.log("cert compare result: " + result);
+
+                if (result === false) {
+                    session.toString();
+                    err = new Error("Server certificate not in our trust list ");
+                }
             } else {
                 err = new Error("??? " + response.responseHeader.serviceResult);
             }
@@ -583,6 +591,7 @@ OPCUAClient.prototype.createSession = function (userIdentityInfo, callback) {
 
     self._createSession(function (err, session) {
         if (err) {
+            console.log("createSession got error");
             callback(err);
         } else {
 
